spec/controllers/students_controller_spec.rb

```diff
@@ -20,4 +22,47 @@
       end
     end
   end
+
+  describe "GET #show" do
+    subject { get :show, id: student }
+
+    it "assigns the requests student to @student" do
+      subject
+      expect(assigns(:student)).to eq(student)
+    end
+
+    it { is_expected.to have_http_status(:success) }
+    it { is_expected.to render_template("students/show") }
+  end
+
+  describe "GET #edit" do
+    subject { get :edit, id: student }
+
+    it "assigns the requests student to @student" do
+      subject
+      expect(assigns(:student)).to eq(student)
+    end
+
+    it { is_expected.to have_http_status(:success) }
+    it { is_expected.to render_template("students/edit") }
+  end
+
+  describe "PUT #update" do
+    subject { put :update, id: student, student: FactoryBot.attributes_for(:student, first_name: "Mike", last_name: "Jones") }
+
+    it "located the requested @contact" do
+      subject
+      expect(assigns(:student)).to eq(student)
+    end
+
+    it "changes @student's attributes" do
+      subject
+      student.reload
+      expect(student.first_name).to eq("Mike")
+      expect(student.last_name).to eq("Jones")
+    end
```

Comment:
What do you think about this approach instead?
```suggestion
    it "changes @student's attributes" do
      is_expected.to change { student.first_name }.to("Mike")
        .and change { student.last_name }.to("Jones)
    end
```
You may still need to call reload on student. I hope it works.

Reply:
I like that approach, I had to fiddle with it to get it to work, but it's definitely an improvement.